cpp/src/Slice/SliceUtil.cpp

```diff
@@ -470,37 +470,43 @@ Slice::checkIdentifier(const string& id)
         name = id;
     }
 
+    bool isValid = true;
     // check the identifier for reserved suffixes
     static const string suffixBlacklist[] = { "Helper", "Holder", "Prx", "Ptr" };
     for(size_t i = 0; i < sizeof(suffixBlacklist) / sizeof(*suffixBlacklist); ++i)
     {
         if(name.find(suffixBlacklist[i], name.size() - suffixBlacklist[i].size()) != string::npos)
         {
-            unit->error("illegal identifier `" + name + "': `" + suffixBlacklist[i] + "' suffix is reserved");
+            unit->error("illegal " + typeName + " identifier `" + name + "': `" + suffixBlacklist[i] + "' suffix is reserved");
```

Comment:
I am wondering if we should replace all the "{typeName} identifier {name}" messages by "{typeName} name {name}" as in:
```
illegal module name FooHelper
illegal interface name Foo__Bar
```
or maybe just:

```
module FooHelper is invalid: the Helper suffix is reserved
interface Foo__Bar is invalid: double underscores are not allowed in identifiers
```

Reply:
For a future PR: should we keep reserving Holder and Ptr?

Reply:
I think that we should keep 'identifier', since that's what they're formally called in the documentation: https://doc.zeroc.com/ice/3.7/the-slice-language/lexical-rules#identifiers

Even if names sounds more natural, I think the compilers should be as formally correct as possible.

Reply:
What about my second suggestion, with does not use name:
```
module FooHelper is invalid: the Helper suffix is reserved
interface Foo__Bar is invalid: double underscores are not allowed in identifiers
```

Reply:
I'm okay with the second set of suggestions, but I don't find it any better or worse than the current comments is all.
```
module FooHelper is invalid: the Helper suffix is reserved
illegal module identifier `FooHelper': Helper suffix is reserved
```
The second one is slightly more verbose, but also explicitly states it's the identifier that's invalid, instead of the module in general. Either way, I think the messages both work. If you much prefer your alternate suggestions though, I can change them obviously.

Reply:
Yes, I prefer my suggested messages!

Reply:
Alright, what about instead of
```
module FooHelper is invalid: the Helper suffix is reserved
```
We do:
```
module FooHelper's identifier is invalid: the Helper suffix is reserved
```
Because I do like the explicitness of stating that it's specifically the _identifier_ that's invalid, and not necessarily the entire module itself...